open-machine-learning-jupyter-book/assignments/deep-learning/gan/gan-introduction.ipynb

```diff
@@ -0,0 +1 @@
2) == 0:\n","            print('[%d/%d][%d/%d] Loss_D: %.4f Loss_G: %.4f D(x): %.4f D(G(z)): %.4f / %.4f'\n","                  % (epoch + 1, epochs, ii+1, len(train_loader),\n","                     errD.item(), errG.item(), D_x, D_G_z1, D_G_z2))\n","            \n","    plot_loss (G_losses, D_losses, epoch)\n","    G_losses = []\n","    D_losses = []\n","    if epoch % 10 == 0:\n","        show_generated_img()\n","\n","    epoch_time.append(time.time()- start)"]},{"cell_type":"code","execution_count":null,"metadata":{"_kg_hide-output":true,"trusted":true},"outputs":[],"source":["print (\">> average EPOCH duration = \", np.mean(epoch_time))"]},{"cell_type":"markdown","metadata":{},"source":["## Generation example"]},{"cell_type":"code","execution_count":null,"metadata":{"_kg_hide-input":true,"trusted":true},"outputs":[],"source":["show_generated_img(7)"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["if not os.path.exists('../output_images'):\n","    os.mkdir('../output_images')\n","    \n","im_batch_size = 50\n","n_images=10000\n","\n","for i_batch in tqdm(range(0, n_images, im_batch_size)):\n","    gen_z = torch.randn(im_batch_size, nz, 1, 1, device=device)\n","    gen_images = netG(gen_z)\n","    images = gen_images.to(\"cpu\").clone().detach()\n","    images = images.numpy().transpose(0, 2, 3, 1)\n","    for i_image in range(gen_images.size(0)):\n","        save_image(gen_images[i_image, :, :, :], os.path.join('../output_images', f'image_{i_batch+i_image:05d}.png'))"]},{"cell_type":"code","execution_count":null,"metadata":{"_kg_hide-input":true,"trusted":true},"outputs":[],"source":["fig = plt.figure(figsize=(25, 16))\n","# display 10 images from each class\n","for i, j in enumerate(images[:32]):\n","    ax = fig.add_subplot(4, 8, i + 1, xticks=[], yticks=[])\n","    plt.imshow(j)"]},{"cell_type":"markdown","metadata":{},"source":["### Submission"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["import shutil\n","shutil.make_archive('images', 'zip', '../output_images')"]},{"cell_type":"markdown","metadata":{},"source":["### Save models"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["torch.save(netG.state_dict(), 'generator.pth')\n","torch.save(netD.state_dict(), 'discriminator.pth')"]},{"cell_type":"markdown","metadata":{},"source":["## Acknowledgement\n","Thanks to [jesucristo](https://www.kaggle.com/jesucristo) for creating [GAN Introduction](https://www.kaggle.com/code/jesucristo/gan-introduction/notebook). It inspired the majority of the content in this article."]}],"metadata":{"kernelspec":{"display_name":"Python 3","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.8.17"}},"nbformat":4,"nbformat_minor":4}
```

Comment:
<p>This code can be copied to the location where the final image is generated, replacing the definition of netG/netD with the code that loads the model and subsequently generates the predicted image.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ocademy-ai/machine-learning/pull/412/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>